unit_tests/test_simulation_config.py

```diff
@@ -20,46 +19,39 @@ def setUp(self) -> None:
     def tearDown(self) -> None:
         os.environ.pop('SIMULATION_CONFIG_FILE_PATH', None)
 
-    def test_simulation_info_returns_the_default_when_no_json_file_path(self):
-        simulation_id, domain_name, websockets_domain_name = get_simulation_config()
-        assert simulation_id == SIMULATION_ID_FROM_ENV
-        assert domain_name == DOMAIN_NAME_FROM_ENV
-        assert websockets_domain_name == WEBSOCKET_DOMAIN_NAME_FROM_ENV
+    def test_get_sim_id_and_domain_names_returns_the_defaults(self):
+        simulation_id, domain_name, websockets_domain_name = get_sim_id_and_domain_names()
+        assert simulation_id == simulation_id_from_env()
+        assert domain_name == domain_name_from_env()
+        assert websockets_domain_name == websocket_domain_name_from_env()
```

Comment:
I think it's great to test this. One caveat is that, with this code, we're not exactly testing that `simulation_id_from_env()` and the others return the values we put in the env. We're just testing that the values returned by `get_sim_id_and_domain_names()` are the same returned by `simulation_id_from_env()` and the others. However:
> we should be "testing the interface, and NOT the implementation" - doing so will allow us to change the implementation without causing our tests to break.

This means that, regardless of `simulation_id_from_env()` and the others, we should test that:
- If we specify env variables, the values returned by `get_sim_id_and_domain_names()` will be those variables
- If we don't specify env variables, the values returned by `get_sim_id_and_domain_names()` will be the default values.

What would you think about something like this?
```suggestion
    def test_get_sim_id_and_domain_names_returns_the_defaults(self):
        os.environ["API_CLIENT_SIMULATION_ID"] = "test-simulation-id"
        os.environ["API_CLIENT_DOMAIN_NAME"] = "test-domain-name"
        os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = "test-websocket-name"
        
        simulation_id, domain_name, websockets_domain_name = get_sim_id_and_domain_names()
        assert simulation_id == "test-simulation-id"
        assert domain_name == "test-domain-name"
        assert websockets_domain_name == "test-websocket-name"
```

In this way, we don't really care about how `get_sim_id_and_domain_names()` is implemented internally. What would you think?